ppafm/common.py

```diff
@@ -119,10 +119,10 @@ def load_ini(self, lines):
             words = line.split()
             if len(words) >= 2:
                 key = words[0]
+                if key[0] == "#":
+                    continue
                 if hasattr(self, key):
                     val = getattr(self, key)
-                    if key[0][0] == "#":
```

Comment:
What if we leave both of these options, before line 120, then it would not matter, if we start with `# someparameter` or `#someparameter` and it also would not matter on the amount of words at all.

Reply:
Anyway this fix will be short-lived as we are wanting to move towards _toml_ .

Reply:
Done in https://github.com/Probe-Particle/ppafm/pull/303/commits/4939276d6d89d758e96d5f5f1738e2904e1835c0

Reply:
And yes, I figured that this will be moved to toml in the future but I was working on some PPSTM things and this came up...

Reply:
Does this work if the comment character is somewhere in the middle of the line? It looks like a space before `#` would break it at least.

I guess the idea is just to ignore everything after the comment character. I think this in the beginning of the loop would do it:
```python
line = line.split("#")[0].strip()
```

Reply:
Agree that ignoring everything after is the effect we want. However, the parser is quite strict and, e.g., for the grid it gets specifically 3 floats after the key

https://github.com/Probe-Particle/ppafm/blob/7c0aebb7a1a10357a2817af4f24bf93c1c096e82/ppafm/common.py#L143-L145

so comment in the middle of the line breaks the parser if there's not enough values.

```sh
  File "/Users/kurkil1/git/ppafm/ppafm/common.py", line 144, in load_ini
    setattr(self, key, [float(words[1]), float(words[2]), float(words[3])])
IndexError: list index out of range
```

Reply:
What would be the expected behaviour if the `params.ini` is not valid, e.g. only 2 values for the grid? Probably possibilities are to revert to default parameters or raise this error.

Reply:
I would say that it needs to raise some error, so that you don't accidentally use the wrong parameters without knowing.

Maybe the error message could be better. We could add a `try-except` block around the `if-elif`-chain and if it runs into an error, print something like `Error parsing parameters on line "[insert offending line]"`.

Reply:
> Does this work if the comment character is somewhere in the middle of the line? It looks like a space before `#` would break it at least.
> 
> I guess the idea is just to ignore everything after the comment character. I think this in the beginning of the loop would do it:
> 
> ```python
> line = line.split("#")[0].strip()
> ```

done in https://github.com/Probe-Particle/ppafm/pull/303/commits/c400f3020729d67b05833060730be96878e59726

> try-except block...

done in https://github.com/Probe-Particle/ppafm/pull/303/commits/c91bf6d3cef7add59b3355185a4ed7a5bf175dfe